### DISCUSSION TITLE
Allow restarting language server?

### DISCUSSION THREAD

**Discussion by jeroenvisser101** [elixir-lsp/vscode-elixir-ls#237] [Ideas]
Like other language server extensions, maybe it would be good to allow to restart the language server, for instance when changing branches or other operations break the language server and require removing `.elixir_ls` and restarting the server to fix.

<img width="596" alt="image" src="https://user-images.githubusercontent.com/1941348/159131622-1f918ac7-ae61-4024-bb85-f3b9c83144a2.png">
<img width="611" alt="image" src="https://user-images.githubusercontent.com/1941348/159131644-f34c5615-a8ff-42e8-957e-ada2db906666.png">
<img width="626" alt="image" src="https://user-images.githubusercontent.com/1941348/159131659-88248928-8ed1-4571-b33f-2d1f79f83fa3.png">

Having something similar as an escape hatch would be helpful!

**Comment by lukaszsamson** [comment]
@jeroenvisser101 it's already there but not released, see https://github.com/elixir-lsp/vscode-elixir-ls/pull/218.

**Comment by jeroenvisser101** [comment]
@lukaszsamson could there also be a command to force-reindex dialyzer? I notice sometimes that when adding a new field to an Ecto schema, all dependent files will have dialyzer errors. To recover, I can delete `.elixir_ls/{dialyzer_tmp,dialyzer_manifest_..._test}`, and restart the language server.

Maybe this warrants a separate issue, if this is not a known bug? I've never tried to reproduce this outside of our work project, but can do if needed :)

**Reply by lukaszsamson** [reply]
Please open a new issue
